addons/rating/models/rating_mixin.py

```diff
@@ -42,12 +47,38 @@ def _compute_rating_percentage_satisfaction(self):
                 grades_per_parent[parent_id]['okay'] += item['__count']
             else:
                 grades_per_parent[parent_id]['bad'] += item['__count']
+            rating_scores_per_parent[parent_id] += rating * item['__count']
 
         # compute percentage per parent
         for record in self:
             repartition = grades_per_parent.get(record.id, default_grades)
-            record.rating_count = sum(repartition.values())
-            record.rating_percentage_satisfaction = repartition['great'] * 100 / sum(repartition.values()) if sum(repartition.values()) else -1
+            rating_count = sum(repartition.values())
+            record.rating_count = rating_count
+            record.rating_percentage_satisfaction = repartition['great'] * 100 / rating_count if rating_count else -1
+            record.rating_avg = rating_scores_per_parent[record.id] / rating_count if rating_count else 0
+            record.rating_avg_percentage = record.rating_avg / 5
+
+    def _search_rating_avg(self, operator, value):
+        if operator not in ['=', '!=', '<=', '<', '>', '>=']:
+            raise NotImplementedError('This operator %s is not supported in this search method.' % operator)
+        if not isinstance(value, (int, float)):
```

Comment:
Is it necessary ? Could we really send something else with those operators ?

Reply:
The goal of those conditions is to be sure the parameters are valid for this search method.
For the `operator` variable, we accept one of those operators ('=', '!=', '<=', '<', '>', '>=') but not 'like' operator for instance.
For the `value` operator is just to be sure we have a number and not a text for instance. 🙂 
After those conditions, we could be sure the parameters are correct for this search method and we can continue the process in this method. 🙂 😇 

We could see those conditions as *an explicit error* for the developer who use the field `rating_avg` in a domain and did a mistake in one of both parameters. 😇

Reply:
Actually I was asking for the value. When using =, '=, <, >, ... on an integer, would you really receive something else ? Isn't it unnecessary defensive code ? In my opinion it adds unnecessary lines, as: through interface it will be correctly set, and through code you will have an error about parsing instead of a "nice error", but in the end this is not something anyone will encounter in a real life scenario.

Reply:
Oh sorry ^^
Indeed, you are right. 😇